src/Type/ArrayType.php

```diff
@@ -112,6 +112,34 @@ public function isSuperTypeOf(Type $type): TrinaryLogic
 		return TrinaryLogic::createNo();
 	}
 
+	public function isSuperTypeOfMixed(Type $type): TrinaryLogic
+	{
+		return $type->isArray()
+			->and($this->isNestedTypeSuperTypeOf($this->getIterableValueType(), $type->getIterableValueType()))
+			->and($this->isNestedTypeSuperTypeOf($this->getIterableKeyType(), $type->getIterableKeyType()));
```

Comment:
could this be optimized using the `lazyAnd` method recently added to TrinaryLogic? if the value type is not a super type then it's not worth checking the keys aswell